nucypher/characters/base.py

```diff
@@ -207,25 +204,15 @@ def __init__(self,
             self.keyring_root = STRANGER
             self.network_middleware = STRANGER
 
-        #
-        # Decentralized
-        #
-        if not federated_only:
-            self._checksum_address = checksum_address  # TODO: Check that this matches TransactingPower
-
-        #
-        # Federated
-        #
-        elif federated_only:
-            try:
-                self._set_checksum_address()
-            except NoSigningPower:
-                self._checksum_address = NO_BLOCKCHAIN_CONNECTION
-            if checksum_address:
-                # We'll take a checksum address, as long as it matches their singing key
-                if not checksum_address == self.checksum_address:
-                    error = "Federated-only Characters derive their address from their Signing key; got {} instead."
-                    raise self.SuspiciousActivity(error.format(checksum_address))
+        # TODO: Figure out when to do this.
+        try:
+            _transacting_power = self._crypto_power.power_ups(TransactingPower)
+        except NoTransactingPower:
+            self._checksum_address = checksum_address
+            if checksum_address is None:
+                assert True  # Hmm, so this Character has no checksum address at all.  Is that what we want?
```

Comment:
Enrico and most Bobs don't need a checksum address. Why is this a matter of concern here?